src/compile/data/parse.ts

```diff
@@ -57,9 +57,12 @@ export function parseTransformArray(head: DataFlowNode, model: Model, ancestorPa
 
       head = new FilterNode(head, model, t.filter);
     } else if (isBin(t)) {
-      head = BinNode.makeFromTransform(head, t, model);
+      const bin = head = BinNode.makeFromTransform(head, t, model);
+
+      for (const field of keys(bin.producedFields())) {
+        ancestorParse.set(field, 'number', false);
+      }
 
-      ancestorParse.set(t.as, 'number', false);
```

Comment:
So, previously only the `as` `string` was being set. Now, both `as` elements(implicit as well as explicit) are set, which changed that example. The svg is unchanged.